src/test/java/net/wasdev/wlp/common/plugins/util/DevUtilTest.java

```diff
@@ -76,12 +85,150 @@ public void testCleanupServerEnvBak() throws Exception {
+        File tempTargetFile = new File(targetDir, "server.xml");
+        Files.write(tempTargetFile.toPath(), "temp".getBytes());
+
+        // verify that files have been properly created
+        assertTrue(tempSrcFile.exists());
+        assertTrue(tempTargetFile.exists());
+
+        util.deleteFile(tempSrcFile, srcDir, targetDir, "server.xml");
+
+        // verify that the target file has been deleted
+        assertFalse(tempTargetFile.exists());
+
+        // clean up
+        if (srcDir.exists()) {
+            FileUtils.deleteDirectory(srcDir);
+        }
+        if (targetDir.exists()) {
+            FileUtils.deleteDirectory(targetDir);
+        }
+    }
+
+    @Test
+    public void testCleanTargetDir() throws Exception {
+        DevUtil util = new DevTestUtil(null, null, null, null, null, false);
+
+        File outputDirectory = Files.createTempDirectory("outputDirectory").toFile();
+        File tempClass = new File(outputDirectory, "temp.class");
+        Files.write(tempClass.toPath(), "temp".getBytes());
+
+        assertTrue(tempClass.exists());
+
+        util.cleanTargetDir(outputDirectory);
+
+        assertFalse(tempClass.exists());
+
+        outputDirectory = Files.createTempDirectory("outputDirectory").toFile();
+        File tempTextFile = new File(outputDirectory, "temp.txt");
+        Files.write(tempTextFile.toPath(), "temp".getBytes());
+
+        assertTrue(tempTextFile.exists());
+
+        util.cleanTargetDir(outputDirectory);
+        assertTrue(outputDirectory.exists());
+
+        if (outputDirectory.exists()) {
+            FileUtils.deleteDirectory(outputDirectory);
+        }
+    }
+
+    @Test
+    public void testGetFileFromConfigDirectory() throws Exception {
+        DevUtil util = new DevTestUtil(null, null, null, this.configDirectory, null, false);
+        
+        File tempTextFile = new File(configDirectory, "temp.txt");
+        Files.write(tempTextFile.toPath(), "temp".getBytes());
+        
+        File configFile = util.getFileFromConfigDirectory("temp.txt");
+        assertTrue(configFile.exists());
+    }
+    
+    @Test
+    public void testDeleteJavaFile() throws Exception {
+        DevUtil util = new DevTestUtil(null, null, null, null, null, false);
+
+        File compileSourceRoot = Files.createTempDirectory("compileSourceRoot").toFile();
+        File javaFile = new File(compileSourceRoot, "temp.java");
+        Files.write(javaFile.toPath(), "temp".getBytes());
+        File classesDir = Files.createTempDirectory("classesDir").toFile();
+        File javaClass = new File(classesDir, "temp.class");
+        Files.write(javaClass.toPath(), "temp".getBytes());
+        
+        assertTrue(javaFile.exists());
+        assertTrue(javaClass.exists());
+        
+        util.deleteJavaFile(javaFile, classesDir, compileSourceRoot);
+        assertFalse(javaClass.exists());
+        
+        if (compileSourceRoot.exists()) {
```

Comment:
If the test fails before this point, the cleanup won't be run.  It's better to cleanup using in a `finally` block or in the tearDown() method.